src/nc_connection.h

```diff
@@ -38,14 +38,44 @@ typedef void (*conn_msgq_t)(struct context *, struct conn *, struct msg *);
 typedef void (*conn_post_connect_t)(struct context *ctx, struct conn *, struct server *server);
 typedef void (*conn_swallow_msg_t)(struct conn *, struct msg *, struct msg *);
 
+/*
+ * There are few kinds of connections, client-bound, server-bound, redis,
+ * memcache, proxy, etc. We uniquely identify a connection kind using a single
+ * enum, otherwise the code becomes a tangle of if's.
+ */
+enum conn_kind {
```

Comment:
@manjuraj Re combinations of connections. One thing is that it eases the "CONN_KIND_AS_STRING(c)" implementation, so it is O(1). Do you recommend a different approach?

@manjuraj re cleanups: I've added a couple of comments to your changes in the config-reload branch, would you please respond.